pkg/controllers/cloudprovideraccount/controller.go

```diff
@@ -151,7 +151,27 @@ func (r *CloudProviderAccountReconciler) processCreateOrUpdate(namespacedName *t
 
 func (r *CloudProviderAccountReconciler) processDelete(namespacedName *types.NamespacedName) error {
 	r.Log.Info("Received request", "account", namespacedName, "operation", "delete")
-	return r.AccManager.RemoveAccount(namespacedName)
+	if err := r.AccManager.RemoveAccount(namespacedName); err != nil {
+		return err
+	}
+
+	// Set error status on dependent CES.
+	cesList := &crdv1alpha1.CloudEntitySelectorList{}
+	if err := r.Client.List(context.TODO(), cesList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	for _, ces := range cesList.Items {
+		if ces.Spec.AccountName == namespacedName.Name && ces.Spec.AccountNamespace == namespacedName.Namespace {
+			r.Log.Info("Set CES error status", "name", ces.Name, "namespace", ces.Namespace)
+			ces.Status.Error = fmt.Sprintf("owner account %v/%v not found", ces.Spec.AccountName, ces.Spec.AccountNamespace)
+
+			if err := r.Client.Status().Update(context.TODO(), &ces); err != nil {
```

Comment:
How would user recover from this scenario?
Lets say he add's the CPA back, then the error on the CES object is no longer valid.

If there is no way to recover then, then the error message has to be rephrased saying,
Account not found for Selector, please delete the selector

Reply:
Yes, I had the same question, I tried out the case too. We can update the message, I don't know what else we can do here.

Reply:
Update: Instead of setting status on CES, we decided to delete CES when the referenced account gets deleted.